Tests/Tests/IndexManagerTests.m

```diff
@@ -839,6 +843,117 @@ - (void)testNewIndexManagerPicksUpDocumentsAddedBeforeEnsureIndexedDuringQuery
+        args = @[@1, @1, @"1-972630218dba7c669957a8567684b7b2" /*, no parent */, @0, @0, 
+                 @"{\"completed\":false,\"description\":\"desc1\"}"];
+        [db executeUpdate:sql withArgumentsInArray:args];
+        
+        sql = @"INSERT INTO revs (sequence, doc_id, revid, parent, current, deleted, json) "
+        @"VALUES (?, ?, ?, ?, ?, ?, ?)";
+        args = @[@2, @1, @"2-4fef73a591b001c590a28e433bc448c3", @1, @1, @0, 
+                 @"{\"completed\":true,\"description\":\"desc2\"}"];
+        [db executeUpdate:sql withArgumentsInArray:args];
+        
+        args = @[@3, @1, @"2-3314e4f9bbce21fa9a111af2ac6a28eb", @1, @0, @0, 
+                 @"{\"completed\":true,\"description\":\"desc3\"}"];
+        [db executeUpdate:sql withArgumentsInArray:args];
+    }];
+    
+    // At this point, try getting each doc to make sure the above is still the right db format
+    XCTAssertNotNil([datastore getDocumentWithId:testDocId 
+                                             rev:@"1-972630218dba7c669957a8567684b7b2" 
+                                           error:nil]);
+    XCTAssertNotNil([datastore getDocumentWithId:testDocId 
+                                             rev:@"2-4fef73a591b001c590a28e433bc448c3" 
+                                           error:nil]);
+    XCTAssertNotNil([datastore getDocumentWithId:testDocId 
+                                             rev:@"2-3314e4f9bbce21fa9a111af2ac6a28eb" 
+                                           error:nil]);
+    
+    // Set up our index
+    CDTIndexManager *im = [[CDTIndexManager alloc] initWithDatastore:datastore error:nil];
+    [im ensureIndexedWithIndexName:@"description" fieldName:@"description" error:nil];
+    
+    // Query for doc, winning rev is 2-4fe as lexigraphically higher than 2-331
+    CDTQueryResult *res1 = [im queryWithDictionary:@{@"description": @"desc2"} error:nil];  
+    XCTAssertEqual(res1.documentIds.count, (NSUInteger)1);
+    XCTAssertEqualObjects(res1.documentIds[0], testDocId);
+    
+    // Resolve conflict by tombstone-ing 2-331
+    [datastore.database.fmdbQueue inDatabase:^(FMDatabase *db) {
+        NSString *sql;
+        NSArray *args;
+        
+        sql = @"INSERT INTO revs (sequence, doc_id, revid, parent, current, deleted, json) "
+        @"VALUES (?, ?, ?, ?, ?, ?, ?)";
+        args = @[@4, @1, @"3-bc3b310ab4ff877104b1aeb53d0b8457", @3, @1, @1, @""];
+        [db executeUpdate:sql withArgumentsInArray:args];
+    }];
+    XCTAssertNotNil([datastore getDocumentWithId:testDocId 
+                                             rev:@"3-bc3b310ab4ff877104b1aeb53d0b8457" 
+                                           error:nil]);
+    
+    // Should still have the same result.
+    CDTQueryResult *res2 = [im queryWithDictionary:@{@"description": @"desc2"} error:nil];
+    XCTAssertEqual(res2.documentIds.count, (NSUInteger)1);
+    XCTAssertEqualObjects(res2.documentIds[0], testDocId);
```

Comment:
I agree with @tomblench asserting on the document revision will make it easier to read